src/commands/api-service/keys-regenerate.ts

```diff
@@ -0,0 +1,80 @@
+import Command, { flags } from "@oclif/command";
+import * as Parser from "@oclif/parser";
+import chalk from "chalk";
+import cli from "cli-ux";
+import { toError } from "fp-ts/lib/Either";
+import { TaskEither, tryCatch } from "fp-ts/lib/TaskEither";
+// tslint:disable-next-line: no-submodule-imports
+import { getRequiredStringEnv } from "io-functions-commons/dist/src/utils/env";
+import fetch from "node-fetch";
+
+export class KeyRegenerate extends Command {
+  public static description = "Regenerate keys associated to service";
+
+  // tslint:disable-next-line: readonly-array
+  public static examples = [
+    `$ io-ops api-service:keys-regenerate  SERVICEID --key_type=PRIMARY_KEY`
+  ];
+
+  // tslint:disable-next-line: readonly-array
+  public static args: Parser.args.IArg[] = [
+    {
+      description: "id of the service",
+      name: "serviceId",
+      required: true
+    }
+  ];
+
+  public static flags = {
+    key_type: flags.string({
+      description: "JSON string rapresentation of a service",
+      required: true,
+      options: ["PRIMARY_KEY", "SECONDARY_KEY"]
+    })
+  };
+
+  public async run(): Promise<void> {
+    // can get args as an object
+    const { args, flags: commandLineFlags } = this.parse(KeyRegenerate);
+    // tslint:disable-next-line: no-console
+    cli.action.start(
+      chalk.blue.bold(`Getting keys for service ${args.serviceId}`),
+      chalk.blueBright.bold("Running"),
+      {
+        stdout: true
+      }
+    );
+    return this.put(args.serviceId, commandLineFlags.key_type)
+      .fold(
+        error => {
+          cli.action.stop(chalk.red(`Error : ${error}`));
+        },
+        result => {
+          cli.action.stop(chalk.green(`Response: ${result}`));
+        }
+      )
+      .run();
+  }
+
+  private put = (
+    serviceId: string,
+    keyType: string
+  ): TaskEither<Error, string> => {
+    return tryCatch(
+      () =>
+        fetch(
+          `${getRequiredStringEnv(
+            "BASE_URL_ADMIN"
+          )}/services/${serviceId}/keys`,
+          {
+            body: JSON.stringify({ key_type: keyType }),
```

Comment:
While it's amendable for get requests (that just print outputs) we should use the generated types for every PUT and POST request so JSON.stringify should become -> MyApiPayloadType.encode({ ... }) or use a decode step like you did in other commands.

otherwsie that would be very difficult to refactor these methods in case the API spec changes (there's no way using JSON.stringify to notice that there's a type mismatch)